tests/src/FunctionalJavascript/Integration/AutosaveFormTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Drupal\Tests\thunder\FunctionalJavascript\Integration;
+
+use Drupal\Tests\thunder\FunctionalJavascript\ThunderFormFieldTestTrait;
+use Drupal\Tests\thunder\FunctionalJavascript\ThunderJavascriptTestBase;
+use Drupal\Tests\thunder\FunctionalJavascript\ThunderParagraphsTestTrait;
+
+/**
+ * Tests the autosave support for nodes in Thunder.
+ *
+ * @group Thunder
+ */
+class AutosaveFormTest extends ThunderJavascriptTestBase {
+
+  use ThunderFormFieldTestTrait;
+  use ThunderParagraphsTestTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Adjust the autosave form submission interval.
+    \Drupal::configFactory()
+      ->getEditable('autosave_form.settings')
+      ->set('interval', 2000)
+      ->save();
+  }
+
+  /**
+   * Tests the autosave functionality in an existing article.
+   */
+  public function testAutosaveInExistingEntity() {
+    $this->drupalGet('node/7/edit');
+    $page = $this->getSession()->getPage();
+
+    // Make some changes.
+    $this->makeFormChanges();
+
+    // Reload the page.
+    $this->drupalGet('node/7/edit');
+
+    // Reject the changes.
+    $this->pressRejectButton();
+    $this->assertEquals([5], $page->findField('field_tags[]')->getValue());
+    $this->assertEquals('Come to DrupalCon New Orleans', $page->findField('title[0][value]')->getValue());
+
+    // Make changes again.
+    $this->makeFormChanges();
+
+    // Reload the page.
+    $this->drupalGet('node/7/edit');
+
+    $this->pressRestoreButton();
+    $this->assertEquals([5, '$ID:Tag2'], $page->findField('field_tags[]')->getValue());
+    $this->assertEquals('New title', $page->findField('title[0][value]')->getValue());
+
+    // Save the article.
+    $this->clickSave();
+
+    // Check some things.
+    $this->assertSession()->pageTextContains('This post is unpublished and will be published');
+    $this->assertSession()->pageTextContains('Awesome quote');
+  }
+
+  /**
+   * Press the restore button.
+   */
+  protected function pressRestoreButton() {
+    $page = $this->getSession()->getPage();
+
+    // Press restore button.
+    $this->assertSession()->waitForText('A version of this page you were editing at');
+    $restore_button = $page->find('css', '.autosave-form-resume-button');
+    $this->assertNotEmpty($restore_button);
```

Comment:
I'm not sure if you have to `assertNotEmpty`, because if it's empty, the test will fail anyway on `$restore_button->press();`